kotlin-test-launcher/gradle-plugin/src/main/kotlin/io/github/codymikol/kotlintestlauncher/plugin/KotlinTestLauncherPlugin.kt

```diff
@@ -0,0 +1,23 @@
+package io.github.codymikol.kotlintestlauncher.plugin
+
+import io.github.codymikol.kotlintestlauncher.plugin.task.KotlinTestLaunch
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.register
+import java.util.UUID
+
+class KotlinTestLauncherPlugin : Plugin<Project> {
+    override fun apply(project: Project) {
+        project.tasks.register<KotlinTestLaunch>("kotlinTestLaunch") {
+            // Depends on Kotlin compilation to use classes
+            dependsOn("compileTestKotlin")
+
+            group = "verification"
+            description = "Run tests across Kotlin frameworks"
+
+            classes.set(project.properties["classes"]?.toString())
+
+            outputFile.convention(project.layout.buildDirectory.file("$name/output-${UUID.randomUUID()}.json"))
```

Comment:
Currently just dumping to a randomly named output JSON file, but we should be good to provide a custom file via `-PoutputFile=/path/to/output/file/here.json`